src/radical/pilot/agent/executing/popen.py

```diff
@@ -213,8 +213,7 @@ def _handle_unit(self, cu):
                             % (str(e), traceback.format_exc())
 
             # Free the Slots, Flee the Flots, Ree the Frots!
-            if cu.get('slots'):
-                self.publish(rpc.AGENT_UNSCHEDULE_PUBSUB, cu)
+            self.publish(rpc.AGENT_UNSCHEDULE_PUBSUB, cu)
```

Comment:
We always have slots here - otherwise the unit would not make it into executing state.